### PR TITLE
chore(deps): Updated deps, fixed linting errors

### PR DESCRIPTION
**Pull request by mountaindude**
- No new features added. 
- npm dependencies updated to latest (as of today) available versions.
- Fixed a few linting errors caused by recent versions of ELint adding a few checks.
- From what I can tell the extension works just as before.

### CODE REVIEW

**Comment by qlikossbuild** [comment]
Hi, and thanks for contributing!

Note: If you are part of Qlik R&D, you can apply to the `open-source` team in our _internal_ GitHub organization, and ignore the rest of these instructions.

Before we can handle your pull request we need you to sign our contributor license agreement.
This is a one-time process and enables you to contribute without hassle to any of the qlik-oss repositories in the future.

And it only takes a few minutes! Here are some useful links to get you started:

* [Sign the CLA](https://github.com/qlik-oss/open-source/blob/master/sign-cla.md#sign-the-cla)
* [Rationale behind the CLA](https://github.com/qlik-oss/open-source/blob/master/cla-rationale.md)
* [Qlik R&D Open Source repository](https://github.com/qlik-oss/open-source#open-source-at-qlik-rd)
* [Official Slack team](http://branch.qlik.com/slack)

Having troubles? Feel free to reach out to us either here or on Slack!

**Note:** Once the CLA is sorted, you should rebase/touch this pull request to trigger an update of this CLA status.

File: package.json [hunk 1]
```diff
@@ -12,27 +12,34 @@
...
   "dependencies": {
-    "vis": "4.21.0"
+    "@egjs/hammerjs": "2.0.17",
+    "acorn": "8.4.1",
+    "keycharm": "0.4.0",
+    "timsort": "0.3.0",
+    "uuid": "8.3.2",
+    "vis-data": "7.1.2",
+    "vis-network": "9.0.5",
+    "vis-util": "5.0.2"
```
Comment (Caele) [inline package.json:43]: What are these new dependencies?

**Comment by Caele** [comment]
Looks like you included some extra properties as well, lets move that to a different PR.

File: src/paint.js [hunk 2]
```diff
@@ -179,7 +183,7 @@ function paint ( $element, layout, qTheme, component ) {
       $("#"+containerId).css('cursor','default');
 
       network.on('select', function (properties) {
-        if (properties.hasOwnProperty("nodes") && component.options.noInteraction !== true) {
+        if (Object.prototype.hasOwnProperty.call(properties, "nodes") && component.options.noInteraction !== true) {
```
Comment (Caele) [inline src/paint.js:186]: This whole `hasOwnProperty` check can be changed to `properties.nodes`.

**Comment by mountaindude** [comment]
Meh - The later changes I did (adding properties and upgrading to later vis.js version) also got included in this PR. Not what was intended.

The idea was to just update deps of the existing extension, not include any new features/properties. Those should go into separate PR, I agree.

I suggest we start over. Feel free to close this PR and I'll send you a new one, with just the first step (updated deps for the existing extension).

**Pull request closed without merging**